autogpt_platform/backend/backend/blocks/ayrshare/post.py

```diff
@@ -0,0 +1,531 @@
+        )
+        if isinstance(post_result, PostError):
+            yield "error", post_result.message
+            return
+        yield "post_result", post_result
+
+
+class PostToPinterestBlock(AyrsharePostBlockBase):
+    """Block for posting to Pinterest."""
+
+    def __init__(self):
+        super().__init__(
+            id="3ca46e05-dbaa-4afb-9e95-5a429c4177e6",
+            description="Post to Pinterest using Ayrshare",
+        )
+
+    def run(
+        self,
+        input_data: AyrsharePostBlockBase.Input,
+        *,
+        profile_key: SecretStr,
+        **kwargs,
+    ) -> BlockOutput:
+        """Post to Pinterest."""
+        if not profile_key:
+            yield "error", "Please Link a social account via Ayrshare"
+            return
+
+        post_result = self._create_post(
+            input_data,
+            [SocialPlatform.PINTEREST],
+            profile_key=profile_key.get_secret_value(),
+        )
+        if isinstance(post_result, PostError):
+            yield "error", post_result.message
+            return
+        yield "post_result", post_result
+
+
+class PostToTikTokBlock(AyrsharePostBlockBase):
+    """Block for posting to TikTok."""
+
+    def __init__(self):
+        super().__init__(
+            id="7faf4b27-96b0-4f05-bf64-e0de54ae74e1",
+            description="Post to TikTok using Ayrshare",
+        )
+
+    def run(
+        self,
+        input_data: AyrsharePostBlockBase.Input,
+        *,
+        profile_key: SecretStr,
+        **kwargs,
+    ) -> BlockOutput:
+        """Post to TikTok."""
+        if not profile_key:
+            yield "error", "Please Link a social account via Ayrshare"
+            return
+
+        post_result = self._create_post(
+            input_data,
+            [SocialPlatform.TIKTOK],
+            profile_key=profile_key.get_secret_value(),
+        )
+        if isinstance(post_result, PostError):
+            yield "error", post_result.message
+            return
+        yield "post_result", post_result
+
+
+class PostToBlueskyBlock(AyrsharePostBlockBase):
+    """Block for posting to Bluesky."""
+
+    def __init__(self):
+        super().__init__(
+            id="cbd52c2a-06d2-43ed-9560-6576cc163283",
+            description="Post to Bluesky using Ayrshare",
+        )
+
+    def run(
+        self,
+        input_data: AyrsharePostBlockBase.Input,
+        *,
+        profile_key: SecretStr,
+        **kwargs,
+    ) -> BlockOutput:
+        """Post to Bluesky."""
+        if not profile_key:
+            yield "error", "Please Link a social account via Ayrshare"
+            return
+
+        post_result = self._create_post(
+            input_data,
+            [SocialPlatform.BLUESKY],
+            profile_key=profile_key.get_secret_value(),
+        )
+        if isinstance(post_result, PostError):
+            yield "error", post_result.message
+            return
+        yield "post_result", post_result
```

Comment:
can we post to all? seems like a useful feature and blcok

Reply:
+1, or a block where you can choose the platforms to post to with a multiselect